third_party/proto/cosmos/base/query/v1beta1/pagination.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+package cosmos.base.query.v1beta1;
+
+option go_package = "github.com/cosmos/cosmos-sdk/types/query";
+
+// PageRequest is to be embedded in gRPC request messages for efficient
+// pagination. Ex:
+//
+//  message SomeRequest {
+//          Foo some_parameter = 1;
+//          PageRequest pagination = 2;
+//  }
+message PageRequest {
+  // key is a value returned in PageResponse.next_key to begin
+  // querying the next page most efficiently. Only one of offset or key
+  // should be set.
+  bytes key = 1;
+
+  // offset is a numeric offset that can be used when key is unavailable.
+  // It is less efficient than using key. Only one of offset or key should
+  // be set.
+  uint64 offset = 2;
```

Comment:
Is it possible to get just the next key instead of using `offset`?

Reply:
Also, why don't we use `oneof` here?

Reply:
This was how the cosmos SDK pagination got implemented. My design I admit, but open to changing it

Reply:
https://github.com/cosmos/cosmos-sdk/issues/6191